### PR TITLE
Feature/Add configs popup

### PR DESCRIPTION
**Pull request by imhmdb**
Adds a nice configs popup to allow users to enable or disable this extension, could be useful in the future on getting rid of the poling by having an option to manually set RTL or have it the automatic way.

Capture

### CODE REVIEW

File: configs.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            }
+        });
+}
+
+function reloadPage(){
+    chrome.tabs.reload();
+}
+
+document.getElementById('toggleBtn').addEventListener('click', function () {
+    if (this.innerText == 'ON') {
```
Comment (obahareth) [inline configs.js:27]: Can we make this toggle based on the saved value (from Chrome storage) rather than the text value?

File: configs.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+let alert = document.getElementById('alert');
+
+function showAlert(){
```
Comment (obahareth) [inline configs.js:3]: Could you give me a little context on why this alert is needed and also why it needs a timeout? Sorry, my brain isn't at its best at the moment 😆

File: configs.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+let alert = document.getElementById('alert');
```
Comment (obahareth) [inline configs.js:1]: I think this could be a const.

```suggestion
const alert = document.getElementById('alert');
```

File: index.js [hunk 4]
```diff
@@ -18,12 +18,16 @@ function notionPageContentIsLoaded() {
...
 }
 
-let pageLoadInterval = setInterval(() => {
-  if (notionPageContentIsLoaded()) {
-    setBlocksDirectionToAuto()
-
-    // TODO: If we can find a cleaner way to handle block additions/removals
-    // then clear out this interval
-    // clearInterval(pageLoadInterval)  
+chrome.storage.sync.get('mode', function (items) {
```
Comment (obahareth) [inline index.js:21]: I think I would prefer reading this config value separately and not tie the logic that's checking if Notion's page content is loaded to it.

Also one of the things I would love to do is make this work cross-browser and that's one of the reasons why I kind of slacked off on this kind of stuff but I don't want to tie the main logic to Chrome's API only if that makes sense. If we could instead have a module that reads configs (I believe there might also be a package/dependency that serves this purpose) and have that process be separate from our Notion-specific logic that would be great.

Comment (imhmdb) [reply index.js:21]: I think using Mozilla's [WebExtension browser API Polyfill](https://github.com/mozilla/webextension-polyfill) is the best way to go at this point for cross-browser support, this is a starter template to use https://github.com/abhijithvijayan/web-extension-starter and this is an example of a cross-browser extension https://github.com/abhijithvijayan/kutt-extension, have a look at them and let me know if we should move forward in migrating to use the template or use the Polyfill directly without the rest of the starter setup.

p.s.: I don't really do JS much but i'm contributing in this project to refresh my js memory a little bit, so sorry if my PR had some weird js code. XD

Comment (obahareth) [reply index.js:21]: No worries man, I've gotten a little bit rusty myself but maintaining [are-you-es5](https://github.com/obahareth/are-you-es5) every once in a while really helps me out!

**Comment by obahareth** [comment]
@all-contributors please add @imhmdb for code, ideas, and platform contributions.

File: configs.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        this.innerText = 'ON';
+        chrome.storage.sync.set({
+            'mode': 'enabled'
+        })
+    }
+    showAlert();
+})
+
+document.getElementById('reloadBtn').addEventListener('click', reloadPage);
+document.addEventListener('DOMContentLoaded', restoreOptions);
```
Comment (Fahme) [inline configs.js:42]: Avoid global event listeners it's a source for memory leaks.

File: configs.html [hunk 6]
```diff
@@ -0,0 +1,59 @@
+<html>
+
+<head>
+    <meta charset="UTF-8">
+    <title>Notion RTL</title>
+    <style>
```
Comment (Fahme) [inline configs.html:6]: Why not moving this styles to a separate styles.css ? maybe would be cleaner ?

File: configs.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+       alert.style.display = 'block'
+    setTimeout(function () {
+        alert.style.display = 'none'
+    }, 20000)
+}
+
+function restoreOptions() {
+    chrome.storage.sync.get(
+        'mode',
+        function (items) {
```
Comment (Fahme) [inline configs.js:13]: I prefer if anonymous/callback functions always have a name or to be separated it into it's new function for easier debugging and readability 

```chrome.storage.sync.get('mode', function onStorageGet() {})``` etc ...

File: configs.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+            if (items.mode === 'enabled') {
+                document.getElementById('toggleBtn').innerText = 'ON';
+            } else {
```
Comment (Fahme) [inline configs.js:16]: Code needs proper indentation for readability ?

File: configs.js (hunk 3 above)
Comment (Fahme) [inline configs.js:1]: Alert is a very common name so use something else or indicate clearly on why we need this alert ?  

OR 

```(function() {})()``` maybe scope the whole code in an IIFE ? 

what do you think ?

File: configs.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+                document.getElementById('toggleBtn').innerText = 'OFF';
+            }
+        });
+}
+
+function reloadPage(){
+    chrome.tabs.reload();
+}
+
+document.getElementById('toggleBtn').addEventListener('click', function () {
```
Comment (Fahme) [inline configs.js:26]: Cache in variable so that you won't have to always look it up ?
 
```document.getElementById('toggleBtn')```

**Review by Fahme — COMMENTED** [review]
Thanks for you work i've added some comments for clean up,  better code readability and some possible pref improvements that can be done.


@obahareth This been a long hanging PR if you want help in rewriting let me know !

File: configs.js (hunk 3 above)
Comment (Fahme) [inline configs.js:1]: This alert is only being used in showAlert so just add it in showAlert scope.

File: index.js (hunk 4 above)
Comment (Fahme) [reply index.js:21]: ```chrome.storage.sync``` should work fine with 

- Chrome
- Opera
- FF
- Possibility (new edge version didn't really test)

but for safari you will need to do some fallback work to use localStorage.

Note for FF: you will need to specify the id of extension in the manifest.js

**Pull request closed without merging**